src/getters/awaitGraphSync.ts

```diff
@@ -0,0 +1,65 @@
+import getGraphClient from '~/getters/getGraphClient'
+import {
+    GetMetadataDocument,
+    GetMetadataQuery,
+    GetMetadataQueryVariables,
+} from '~/generated/gql/network'
+
+const SESSION_STORAGE_KEY = 'StreamrHubLastBlockNumber'
+
+export const saveLastBlockNumber = (blockNumber: number): void => {
+    window.sessionStorage.setItem(SESSION_STORAGE_KEY, JSON.stringify({ blockNumber }))
+}
+export const awaitGraphSync = async (): Promise<void> => {
```

Comment:
Hmm, we probably should not use the word `await` in the function name. Is `waitForGraphSync` better?

Reply:
good point!

Reply:
I'd really prefer this utility to be just about waiting for a block number. No side effects like storage management.

```ts
async function waitForIndexedBlock(blockNumber: number) {}
```

I like this signature. No need to mention Graph, I guess. All we care about is indexing.

Reply:
@pmichalowski remind me, is session store erased when you close a tab? Because if that's the case when let's store the index in memory.